docs/content/other-languages.md

```diff
@@ -34,3 +34,71 @@ build and run Spin components in that programming language
 
 > Make sure to check out [a more complex Spin application with components built
 in multiple programming languages](https://github.com/fermyon/spin-kitchensink/).
+
+## AssemblyScript
+
+[AssemblyScript](https://www.assemblyscript.org/) is a TypeScript-based language that compiles directly to WebAssembly.
+AssemblyScript has WASI/Wagi support, and so can be used with Spin.
+
+- The [AssemblyScript entry in the Wasm Language Guide](https://www.fermyon.com/wasm-languages/assemblyscript) includes a full example
+- The [Spin Kitchen Sink](https://github.com/fermyon/spin-kitchensink) repo has an AssemblyScript demo
+- An [example AssemblyScript app](https://github.com/deislabs/hello-wagi-as) designed for Wagi runs on Spin
+
+## C/C++
+
+C and C++ are both broadly supported in the WebAssembly ecosystem. WASI/Wagi support means that both can be used to write Spin apps.
+
+- The [C entry in the Wasm Language Guide](https://www.fermyon.com/wasm-languages/c-lang) has examples.
+- The [C++ entry in the Wasm Language Guide](https://www.fermyon.com/wasm-languages/cpp) has specific caveats for writing C++ (like exception handling)
+- The [yo-wasm](https://github.com/deislabs/yo-wasm) project makes setting up C easier.
+
+## C# and .NET languages
+
+.NET has experimental support for WASI, so many (if not all) .NET languages, including C# and F#, can be used to write Spin applications.
+
+- The [C# entry in the Wasm Language Guide](https://www.fermyon.com/wasm-languages/c-sharp) has a full example.
+- The [Spin Kitchen Sink repo](https://github.com/fermyon/spin-kitchensink) has two C# examples and one F# example.
+
+## Grain
+
+[Grain](https://grain-lang.org/), a new functional programming language, has WASI/Wagi support and can be used to write Spin apps.
+
+- The [Grain entry in the Wasm Language Guide](https://www.fermyon.com/wasm-languages/grain) has details
+- A simple [Hello World example](https://github.com/deislabs/hello-wagi-grain) shows how to use Grain
+- For a production-quality example. the [Wagi Fileserver](https://github.com/deislabs/wagi-fileserver) is written in Grain
+
+## Haskell
```

Comment:
I would counsel against mentioning this unless you have had a lot more luck building it than I have.  Running one of their generated modules under the `wasmtime` CLI gives me _unknown import: `StablePtr::newStablePtr` has not been defined_.  Supposedly the Wasm can be run under Node but that just gives me a different error.

Or we could say "they _claim_ this works but we have never tried it."  Just something super non committal!

Reply:
I tried it... and I can't figure out how to make it work. So I removed.